etl/src/main/java/org/pih/hivmigration/etl/sql/SqlMigrator.java

```diff
@@ -211,17 +213,10 @@ void loadFromOracleToMySql(String targetStatement, String sourceQuery, int rowLi
     }
 
     public void clearTable(String tableName) throws SQLException {
-        clearTable(tableName, false);
-    }
-
-    public void clearTable(String tableName, boolean disableForeignKeyChecks) throws SQLException {
-        executeMysql((disableForeignKeyChecks ? "SET FOREIGN_KEY_CHECKS=0;\n" : "SET FOREIGN_KEY_CHECKS=1;\n")
-                + "DROP TABLE IF EXISTS move_tmp_old;\n"
-                + "DROP TABLE IF EXISTS move_tmp;\n"
-                + "CREATE TABLE move_tmp LIKE " + tableName + ";\n"
-                + "RENAME TABLE " + tableName + " TO move_tmp_old, move_tmp TO " + tableName + ";\n"
-                + "DROP TABLE move_tmp_old;\n"
-                + (disableForeignKeyChecks ? "SET FOREIGN_KEY_CHECKS=1;\n" : ""));
+        // TODO: Investigate whether this can be done super fast using `SHOW CREATE TABLE`
```

Comment:
Deleting the way I'm doing it here takes a some minutes to empty a table with millions of entries. It would definitely be faster to be able to drop and re-create, if we could reliably re-create the foreign keys as well. We'll see how much of a problem this is once we start working on obs. My guess is that it generally won't be a problem since we'll generally be using the `limit` parameter while developing.

Reply:
Sorry, that's not what I meant.  I meant, why are you doing a LIMIT 100000 in your DELETE, and running this multiple times?  I would trust the MySQL optimizer to handle this, and just do `"DELETE FROM " + tableName` one time.

Reply:
For tables with millions of entries, the connection times out.

Reply:
Maybe it wasn't connection timeout... anyway, it wouldn't clear the Encounters table all at once, I forget exactly why.

Reply:
Maybe try to use the TRUNCATE command instead and see how that works @brandones ?

Reply:
I can't believe that TRUNCATE never came up in my searching for how to delete all entries of a mysql table quickly.

Just tried it and unfortunately you can't TRUNCATE a table that has foreign key relationships defined (even if the referencing tables are empty, i.e. no actual foreign keys exist). Is this an acceptable use of `SET FOREIGN_KEY_CHECKS=0`?

Reply:
Seems an acceptable use of set foreign key checks = 0 to me...